src/System.Numerics.Tensors/System/Numerics/Tensors/Tensor.Operations.tt

```diff
@@ -10,14 +10,14 @@ namespace System.Numerics.Tensors
     public static partial class Tensor
     { 
 <# foreach (MethodConfiguration method in methodConfiguration) { #>
-        public static <#= method.GetGenericResultMethodSignature("Tensor", "T")#>
+        internal static <#= method.GetGenericResultMethodSignature("Tensor", "T")#>
```

Comment:
There were a couple arithmetic things (like `Zero`) that are used all over the place (even in constructors).

I found it easier to just make the operations internal, than to move them out of the assembly entirely.

Reply:
Was it too expensive to split out?  Typically I don't like IVT in a case like unless we are planning on eventually shipping arithmetic from here.  I was expecting we'd just move the static methods to the test and make sure they worked over the public API.

Reply:
There is a lot of coupling and it isn't definitely clear that everything should be removed or duplicated, etc. Things like `Tensor.CreateIdentity` depend on arithmetic, as well as various constructors, getters/setters (including things like `GetValue` and `SetValue`), etc. You also wouldn't be able to get rid of the IVTs due to dependencies on the helper utilities.

Reply:
(most of the coupling is to `TensorArithmetic.Zero` and `TensorArithmetic.One`)

Reply:
Can we just leave `TensorArithmetic.Zero` and `TensorArithmetic.One` in the library then? I don't understand why'd we want the other dead code in the library.

Reply:
It would require splitting the T4 template and generated interfaces. I think that it's probably more effort than its worth for getting the package published initially.

What I would really like to do is spend a bit more time looking at the type and understanding this implementation of Tensors and some of the design decisions behind it (I did talk with @ericstj briefly).  I don't think the design needs to change a lot for V1, but I do think there might be a couple core things that we should probably think more deeply about, especially if we are going to make this first class and be part of CoreFX.

Some initial thoughts are:
* reference types, in general, are a bad choice for any interop (IMO) because they require marshaling.
* It seems like quite a bit of this was also designed before we did a lot of the perf work (such as ref structs, hardware intrinsics, etc)
* for the `DenseTensor` case, we are only holding a small struct; and we are likewise only holding a single reference type field for the `SparseTensor` case. So, having `structs` with backing interfaces that define the contract seems much more appropriate at first glance. If we are going to have this be a first class citizen (this also allows things like generic-specialization, if users care about it).

Reply:
I absolutely want to think more about how to improve this type, but I don't see how any of this blocks us from making the arithmetic change.  For the other ideas you list here: please open up issues and we can have those conversations, but we shouldn't be having them here in this PR.

Reply:
Split the code apart.